### ISSUE TITLE
SHOW TAG KEYS FROM measurement fails in non default retention policy when autogen (or the default) is empty

### ISSUE THREAD

**Issue by mkontula** [influxdata/influxdb#9819]
### Bug report

__System info:__ InfluxDB 1.3.5, Ubuntu 16.10 (GNU/Linux 4.8.0-59-generic x86_64)

__Steps to reproduce:__

1. CREATE DATABASE repro
2. use repro  
Using database repro
3. CREATE RETENTION POLICY "rp_non_default" ON "repro" DURATION 60d REPLICATION 1
4. INSERT INTO rp_non_default measure_x,tag1=value,tag2=value value1=10,value2=20
5. SELECT * FROM measure_x  
(empty, as it should)
6. SELECT * FROM rp_non_default.measure_x  
```
time                tag1  tag2  value1 value2
----                ----  ----  ------ ------
1525867503763251079 value value 10     20
```
7. SHOW TAG KEYS FROM measure_x
(empty, what?)
8. SHOW TAG KEYS FROM rp_non_default.measure_x
```
name: measure_x
tagKey
------
tag1
tag2
```
9. INSERT INTO autogen measure_y,tagx=valuex,tagy=valuey valuex=10,valuey=20
10. SHOW TAG KEYS FROM measure_x
```
name: measure_x
tagKey
------
tag1
tag2
```
_What, again? Why is it working now after step 9 has been made and not in step 7.?_

__Expected behavior:__ After inserting data to an empty database to non default retention policy, SHOW TAG KEYS FROM measure_x should return the keys.

__Actual behavior:__ After inserting data to an empty database to non default retention policy, SHOW TAG KEYS FROM measure_x (without stating the retention policy name) returns nothing. After inserting completely non related measurement to autogen (default rp in empty db) SHOW TAG KEYS FROM measure_x (without stating the retention policy name) returns tag keys. 

Inserting the data to the default retention policy seems to change functionality. I think this shouldn't be the case and SHOW TAGS FROM <measurement> should return the tag keys even if there's nothing in the default retention policy.

__Additional info:__ 
This came up when using Vibrant's InfluxDB client for C#. The library uses SHOW TAG KEYS query to figure out which columns are tags and which are fields. When running against an empty database, we got erroneous results since the SHOW TAG KEYS query didn't return anything. However, with the steps above this is reproducible from shell as well.

**Comment by russorat** [comment]
closing due to age. Please upgrade to the latest influxdb and open a new issue if the problem persists.
